dll/win32/browseui/explorerband.cpp

```diff
@@ -591,46 +524,176 @@ LRESULT CExplorerBand::OnShellEvent(UINT uMsg, WPARAM wParam, LPARAM lParam, BOO
     return 0;
 }
 
+BOOL
+CExplorerBand::IsTreeItemInEnum(
+    HTREEITEM hItem,
+    IEnumIDList *pEnum)
+{
+    NodeInfo* pNodeInfo = GetNodeInfo(hItem);
+    if (!pNodeInfo)
+        return FALSE;
+
+    pEnum->Reset();
+
+    CComHeapPtr<ITEMIDLIST_RELATIVE> pidlTemp;
+    while (pEnum->Next(1, &pidlTemp, NULL) == S_OK)
+    {
+        if (ILIsEqual(pidlTemp, pNodeInfo->relativePidl))
+            return TRUE;
+
+        pidlTemp.Free();
+    }
+
+    return FALSE;
+}
+
+BOOL
+CExplorerBand::TreeItemHasThisChild(
+    HTREEITEM hItem,
+    PITEMID_CHILD pidlChild)
+{
+    for (hItem = TreeView_GetChild(m_hWnd, hItem); hItem;
+         hItem = TreeView_GetNextSibling(m_hWnd, hItem))
+    {
+        NodeInfo* pNodeInfo = GetNodeInfo(hItem);
+        if (ILIsEqual(pNodeInfo->relativePidl, pidlChild))
+            return TRUE;
+    }
+
+    return FALSE;
+}
+
+HRESULT CExplorerBand::GetItemEnum(CComPtr<IEnumIDList>& pEnum, HTREEITEM hItem)
+{
+    NodeInfo* pNodeInfo = GetNodeInfo(hItem);
+
+    CComPtr<IShellFolder> psfDesktop;
+    HRESULT hr = SHGetDesktopFolder(&psfDesktop);
+    if (FAILED(hr))
+        return hr;
+
+    CComPtr<IShellFolder> pFolder;
+    hr = psfDesktop->BindToObject(pNodeInfo->absolutePidl, NULL, IID_PPV_ARG(IShellFolder, &pFolder));
+    if (FAILED(hr))
+        return hr;
+
+    return pFolder->EnumObjects(NULL, SHCONTF_FOLDERS, &pEnum);
+}
+
+BOOL CExplorerBand::ItemHasAnyChild(HTREEITEM hItem)
+{
+    CComPtr<IEnumIDList> pEnum;
+    HRESULT hr = GetItemEnum(pEnum, hItem);
+    if (FAILED(hr))
+        return FALSE;
+
+    CComHeapPtr<ITEMIDLIST_RELATIVE> pidlTemp;
+    hr = pEnum->Next(1, &pidlTemp, NULL);
+    return SUCCEEDED(hr);
+}
+
+void CExplorerBand::RefreshRecurse(HTREEITEM hTarget)
+{
+    NodeInfo* pNodeInfo = GetNodeInfo(hTarget);
+
+    CComPtr<IEnumIDList> pEnum;
+    HRESULT hrEnum = GetItemEnum(pEnum, hTarget);
+
+    // Delete zombie items
+    HTREEITEM hItem, hNextItem;
+    for (hItem = TreeView_GetChild(m_hWnd, hTarget); hItem; hItem = hNextItem)
+    {
+        hNextItem = TreeView_GetNextSibling(m_hWnd, hItem);
+
+        if (SUCCEEDED(hrEnum) && !IsTreeItemInEnum(hItem, pEnum))
+        {
+            DeleteItem(hItem);
+            continue;
```

Comment:
Since there's nothing else that follows in the for-loop, this continue looks redundant.

Reply:
Fixed.